edx/edx-enterprise
# [ENT-61] Move data sharing consent to Enterprise app

This PR takes the previously-created SSO data sharing consent logic and moves it to live in the Enterprise app.

**JIRA tickets**: Implements ENT-61

**Dependencies**: Requires a specific branch of edx-platform; [available here](https://github.com/open-craft/edx-platform/tree/haikuginger/final-consent-hooks)

**Sandbox URL**: http://149.202.178.151 (use staff/edx username); sandbox currently at 5cec668.

**Testing instructions**:

1. In the sandbox admin, verify that the TestShib SAML IdP is linked to an EnterpriseCustomer, and that that EC does not require data sharing consent.
2. Register a new user, using the TestShib SAML IdP. Verify that the process is identical to the standard SSO process.
3. Disconnect that user from the IdP and sign out.
4. Modify the EC setting to request data sharing consent.
5. Register a new user, using the TestShib SAML IdP, noting that a checkbox for consent is present, checking it.
7. Disconnect this user from the IdP and sign out.
8. Register a new user, using the TestShib SAML IdP, noting that a checkbox for consent is present, but leaving it blank.
9. Sign out of this user, but do not disconnect it from the IdP.
11. In the admin panel, set the EC to require data sharing consent.
12. Attempt to sign in, using the TestShib SAML user account you registered previously. Note that it redirects you to a page stating that additional permissions are required. Do not grant permission, but click Submit, noting that you are logged out.
13. Again attempt to sign in, using the TestShib SAML user account you registered previously. Again note that you are redirected to a page stating that additional permissions are required. Check the box granting permissions and click Submit, noting that you are logged in successfully.
14. Disconnect the user from the IdP and sign out.
15. Attempt to register a new user using the TestShib SAML IdP. Note that the checkbox for data sharing consent is present; note that if you attempt to register a user without checking it, you encounter an error message.
16. Note that once registered, the OpenEdX platform behaves normally.

**Author notes and concerns**:

1. In functionality, this is very similar to what we already have. It does not currently include logic to create an EnterpriseCustomerUser object; we'll likely want to make the relevant changes as a follow-up.

2. The sandbox must be accessed directly via IP. We're using a load balancer to obtain HTTPS, and the domain name is pointed at that LB; however, Django is not aware of that and believes that everything is happening over HTTP. This has resulted in a failure where the redirect URL Django creates for us has an HTTP scheme, and the LB redirects the SAML postback to an HTTPS GET, dropping essential information. Long story short, it's fixable, and not an issue with the codebase, but for right now, just use the sandbox by IP address.

**Reviewers**
- [ ] @saleem-latif 

**Merge checklist:**

**Post merge:**
      finished.
